src/index.ts

```diff
@@ -127,59 +131,39 @@ export class I18N {
             return key;
         }
 
-        if (params) {
-            const count = Number(params.count);
-
-            if (Array.isArray(keyValue)) {
-                if (Number.isNaN(count)) {
-                    this.warn(
-                        'Missing params.count for key.',
-                        keysetName,
-                        key,
-                    );
+        if (Array.isArray(keyValue)) {
+            const count = Number(params?.count);
 
-                    return key;
-                }
-
-                const lastNumber = count % 10;
-                const lastNumbers = count % 100;
-                let index;
-
-                if (count === 0) {
-                    index = 3;
-                } else if (lastNumber === 1 && lastNumbers !== 11) {
-                    index = 0;
-                } else if ((lastNumber > 1 && lastNumber < 5) && (lastNumbers < 10 || lastNumbers > 20)) {
-                    index = 1;
-                } else {
-                    index = 2;
-                }
+            if (Number.isNaN(count)) {
+                this.warn(
+                    'Missing params.count for key.',
+                    keysetName,
+                    key,
+                );
 
-                result = keyValue[index];
-            } else {
-                result = keyValue;
+                return key;
             }
 
-            Object.keys(params).forEach((param) => {
-                let replacer = params[param];
-                if (typeof replacer === 'string' && replacer.indexOf('$') > -1) {
-                    // заменить все одиночные символы '$' на '$$'
-                    replacer = replacer.replace(/(?:([^$])\$|^\$)(?!\$)/g, '$1$$$$');
-                }
+            result = pluralize(keyValue, count);
 
-                result = (result || '') as string;
-                // eslint-disable-next-line security/detect-non-literal-regexp
-                result = result.replace(new RegExp(`({{${param}}})`, 'g'), replacer)
-            });
+            if (keyValue[Plural.None] === undefined) {
+                this.warn('Missing key for 0', keysetName, key);
+            }
         } else {
             result = keyValue;
         }
 
+        if (!params) {
+            return result;
+        }
+
+        result = replaceParams(result, params);
```

Comment:
Давай возвращать `result` в одном месте
```suggestion
        if (params) {
            result = replaceParams(result, params);
        }
```